crate_universe/private/crates_repository.bzl

```diff
@@ -82,10 +84,15 @@ def _crates_repository_impl(repository_ctx):
         repository_dir = repository_ctx.path("."),
         cargo = cargo_path,
         rustc = rustc_path,
+        paths_to_track_file = paths_to_track_file,
         # sysroot = tools.sysroot,
         **kwargs
     )
 
+    paths_to_track = json.decode(repository_ctx.read(paths_to_track_file))
+    for path in paths_to_track:
+        repository_ctx.read(path)
```

Comment:
Should this instead use
https://bazel.build/rules/lib/builtins/repository_ctx#watch
or
https://bazel.build/rules/lib/builtins/repository_ctx#watch_tree
?

Reply:
I wish! `watch` errors on out-of-workspace files, whereas `read` does implicitly watch them. Added comments!